.github/workflows/osx.yml

```diff
@@ -43,7 +43,8 @@ jobs:
           -DSPARROW_IPC_BUILD_SHARED=${{ matrix.build_shared }} \
           -DSPARROW_IPC_BUILD_TESTS=ON \
           -DSPARROW_IPC_BUILD_INTEGRATION_TESTS=ON \
-          -DSPARROW_IPC_BUILD_EXAMPLES=ON
+          -DSPARROW_IPC_BUILD_EXAMPLES=ON \
+          -DUSE_DATE_POLYFILL=ON
```

Comment:
I don't think this has any effect as `sparrow` is already built with `USE_DATE_POLYFILL=ON` on unix in `conda-forge`, it's just needs to be set in the build from source case for these platforms, and that should be done when fetching `sparrow` in `cmake/external_dependencies.cmake`.